staging/src/k8s.io/component-base/zpages/statusz/statusz.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusz
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/munnerz/goautoneg"
+
+	"k8s.io/klog/v2"
+)
+
+var (
+	delimiters              = []string{":", ": ", "=", " "}
+	errUnsupportedMediaType = fmt.Errorf("media type not acceptable, must be: text/plain")
+)
+
+const (
+	headerFmt = `
+%s statusz
+Warning: This endpoint is not meant to be machine parseable, has no formatting compatibility guarantees and is for debugging purposes only.
+`
+
+	dataTemplate = `
+Started{{.Delim}} {{.StartTime}}
+Up{{.Delim}} {{.Uptime}}
+Go version{{.Delim}} {{.GoVersion}}
+Binary version{{.Delim}} {{.BinaryVersion}}
+Emulation version{{.Delim}} {{.EmulationVersion}}
+Minimum Compatibility version{{.Delim}} {{.MinCompatibilityVersion}}
```

Comment:
I mentioned it here https://github.com/kubernetes/kubernetes/pull/125577#discussion_r1829742461, but yeah it would probably be best not to expose it at this time.

Reply:
OK, I removed min compatibility version from the response.